tests/controller/test_server_socket.py

```diff
@@ -0,0 +1,62 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+from flask import Flask
+
+from moviebot.controller.server_socket import ChatNamespace, app, socketio
+
+
+@pytest.fixture
+def flask_app() -> Flask:
+    app.testing = True
+    return app
+
+
+@pytest.fixture
+def client(flask_app):
+    socketio.on_namespace(ChatNamespace("/"))
+    return socketio.test_client(flask_app)
+
+
+@pytest.mark.parametrize(
+    "event, expected_event",
+    [
+        ("register", "registration_response"),
+        ("login", "login_response"),
+    ],
+)
+def test_handle_authentication_empty_fields(client, event, expected_event):
+    """Test _handle_authentication method for registration with empty fields."""
```

Comment:
Nit: if I understand correctly it also tests logging in.

Reply:
Done